### ISSUE TITLE
FlatList re-renders and turns blank when text input in a row scrolls horizontally

### ISSUE THREAD

**Issue by davidfant** [necolas/react-native-web#1800]
**The problem**
When having scrollable components inside of a scrollable component (e.g. `ScrollView`, `FlatList`, `SectionList`), the component's `onScroll` is triggered when a child view is scrolled.

This causes issues with virtualized lists. Example:
1. Scroll down in a FlatList
2. One FlatList row has a text input
3. Scroll horizontally in that text input
4. The text input's scroll calls onScroll in VirtualizedList (which FlatList uses), which causes the VirtualizedList to re-render using the scroll offset of the text input
5. This causes the FlatList to be blank because it's rendering items further up, based on the text input's scroll offset

**How to reproduce**
Simplified test case: https://codesandbox.io/s/objective-pascal-4mhic

Steps to reproduce:
1. Scroll down 20-30 text inputs in the list
2. Write in one of the inputs until the text overflows
3. This leads to the input scrolling horizontally, which triggers the `FlatList`'s `onScroll`
4. This also triggers the `FlatList` to re-render, using the input's scroll offset to calculate the `FlatList`'s renderable items
5. The FlatList will be "blank", but renders the rows at the top of the list

**Expected behavior**
FlatList shouldn't re-render when a child component scrolls

**Environment (include versions). Did this work in previous versions?**
* React Native for Web (version): 0.14.7
* React (version): 16.9.0
* Browser: Brave

The same bug is reproducible in react-native-web 0.13.4

**Additional context**
We've patched this locally by applying the following diff:
```diff
diff --git a/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js b/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
index defde68..4e60592 100644
--- a/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
+++ b/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
@@ -8,6 +8,7 @@
  */
 import * as React from 'react';
 import { forwardRef, useRef } from 'react';
+import { findDOMNode } from 'react-dom';
 import StyleSheet from '../StyleSheet';
 import View from '../View';
 
@@ -84,6 +85,12 @@ var ScrollViewBase = forwardRef(function (props, forwardedRef) {
     scrollLastTick: 0
   });
   var scrollTimeout = useRef(null);
+  var ref = useRef(null);
+
+  function handleRef(_ref) {
+    ref.current = _ref;
+    forwardedRef(_ref);
+  }
 
   function createPreventableScrollHandler(handler) {
     return function (e) {
@@ -123,10 +130,18 @@ var ScrollViewBase = forwardRef(function (props, forwardedRef) {
     scrollState.current.scrollLastTick = Date.now();
   }
 
+  function isComponentTheEventTarget(event) {
+    try {
+      return event.target === findDOMNode(ref.current);
+    } catch (error) {
+      return false;
+    }
+  }
+
   function handleScrollTick(e) {
     scrollState.current.scrollLastTick = Date.now();
 
-    if (onScroll) {
+    if (onScroll && isComponentTheEventTarget(e)) {
       onScroll(normalizeScrollEvent(e));
     }
   }
@@ -134,7 +149,7 @@ var ScrollViewBase = forwardRef(function (props, forwardedRef) {
   function handleScrollEnd(e) {
     scrollState.current.isScrolling = false;
 
-    if (onScroll) {
+    if (onScroll && isComponentTheEventTarget(e)) {
       onScroll(normalizeScrollEvent(e));
     }
   }
@@ -153,7 +168,7 @@ var ScrollViewBase = forwardRef(function (props, forwardedRef) {
     onTouchMove: createPreventableScrollHandler(onTouchMove),
     onWheel: createPreventableScrollHandler(onWheel),
     pointerEvents: pointerEvents,
-    ref: forwardedRef,
+    ref: handleRef,
     style: [style, !scrollEnabled && styles.scrollDisabled, hideScrollbar && styles.hideScrollbar],
     testID: testID
   });
diff --git a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
```

Basically, before triggering a ScrollView's `onScroll`, we make sure that the event target is the current view. If this is an acceptable solution, I'm more than happy to submit a PR.

**Comment by necolas** [comment]
Thanks. I pushed a fix to master
